app/(sign-up)/sign-up.js

```diff
@@ -0,0 +1,328 @@
+    setMobileNumber(phoneNumber);
+    if (phoneNumberRegex.test(phoneNumber)) {
+      setMobileError("");
+    } else {
+      setMobileError("Invalid Mobile Number!");
+    }
+  };
+
+  const handleEmailChange = (text) => {
+    setEmail(text);
+    // Validate email
+    if (text.trim() === "" || emailRegex.test(text)) {
+      setEmailError("");
+    } else {
+      setEmailError("Invalid email address!");
+    }
+  };
+
+  return (
+    <View style={styles.container}>
+      <SafeAreaView style={styles.safeView}>
+        <ScrollView style={styles.scroll} showsVerticalScrollIndicator={false}>
+          {/* Title and Logo */}
+          <View style={styles.titleContainer}>
+            <View style={styles.textContainer}>
+              <Text style={styles.title}>Create an Account</Text>
+              <Text style={styles.subtitle}>
+                Create an account first before using{" "}
+                <Text style={{ color: "#9DB2BF", fontWeight: "bold" }}>
+                  Rent A Car{" "}
+                </Text>
+                Service
+              </Text>
+            </View>
+            <View style={styles.logoContainer}>
+              <Image
+                source={require("../../assets/images/logo.png")}
+                style={styles.logo}
+              />
+            </View>
+          </View>
+          {/* Sign-Up Fields */}
+          <View style={styles.inputContainer}>
+            <Text style={styles.label}>First Name</Text>
+            <TextInput
+              style={styles.input}
+              value={firstName}
+              onChangeText={(text) => setFirstName(text)}
+            />
+
+            <Text style={styles.label}>Last Name</Text>
+            <TextInput
+              style={styles.input}
+              value={lastName}
+              onChangeText={(text) => setLastName(text)}
+            />
+
+            <Text style={styles.label}>Address</Text>
+            <TextInput
+              style={styles.input}
+              value={address}
+              onChangeText={(text) => setAddress(text)}
+            />
+
+            <Text style={styles.label}>Email</Text>
+            <TextInput
+              style={styles.input}
+              autoComplete="email"
+              autoCapitalize="none"
+              keyboardType="email-address"
+              value={email}
+              onChangeText={handleEmailChange}
+            />
+            {!!emailError && <Text style={styles.errorText}>{emailError}</Text>}
+
+            <Text style={styles.label}>Mobile Number</Text>
+            <TextInput
+              style={styles.input}
+              keyboardType="phone-pad"
+              value={mobileNumber}
+              onChangeText={isValidPhoneNumber}
+            />
+            {!!MobileError && (
+              <Text style={styles.errorText}>{MobileError}</Text>
+            )}
+
+            <Text style={styles.label}>Password</Text>
+            <TextInput
```

Comment:
you can make this a component doi dira mo butang ang input tapos ang error handler mo para pwd mo ma reuse, gamiton molang nga component